lib/workers/repository/process/write.ts

```diff
@@ -59,22 +60,30 @@ export async function writeUpdates(
 
   for (const branch of branches) {
     const { baseBranch, branchName } = branch;
+    const branchSha = getBranchCommit(branchName)!;
     const meta: Record<string, string> = { branch: branchName };
     if (config.baseBranches?.length && baseBranch) {
       meta['baseBranch'] = baseBranch;
     }
     addMeta(meta);
     const branchExisted = branchExists(branchName);
-    let branchCache = {} as BranchCache;
-    if (branchExisted && config.repositoryCache === 'enabled') {
-      branchCache =
-        cachedBranches?.find((br) => br.branchName === branchName) ??
-        ({} as BranchCache);
+    let branchState =
+      cachedBranches.find((br) => br.branchName === branchName) ??
+      ({} as BranchCache);
 
-      if (Object.keys(branchCache).length === 0) {
+    if (Object.keys(branchState).length === 0) {
```

Comment:
This logic seems unnecessarily complicated. First you initialize it as `{}` and then immediately after you check whether it's empty. Easier to just not initialize it and check if the value is falsey.

Reply:
I had updated it but forgot to push that code I think, my bad😅